game/main.c

```diff
@@ -3298,24 +3301,26 @@ void display_lines_cleared()
 }
 
 void display_score()
-{
-	static unsigned long temp_score;
-	static unsigned char i;
-
-	temp_score = cur_score; //cur_score;
+{	
+	// if the score was updated since reset
+	if (cur_score != 0)
+	{
+		static unsigned long temp_score;
+		static unsigned char i;
 
-	// clear out any old score.
-	multi_vram_buffer_horz("      ", 6, get_ppu_addr(cur_nt, 0, 6 << 3));
+		temp_score = cur_score; //cur_score;
 
-	i = 0;
-	while (temp_score != 0)
-	{
-		unsigned char digit = temp_score % 10;
-		one_vram_buffer('0' + digit, get_ppu_addr(cur_nt, (6 << 3) - (i << 3), 6 << 3));
+		i = 0;
+		while (temp_score != 0)
+		{
+			unsigned char digit = temp_score % 10;
```

Comment:
Unrelated to this change, but this temp should be turned into a static to improve perf.

Reply:
I'm referring to "digit"

Reply:
Done as per suggestion: https://github.com/tmagri/mbh-firstnes/commit/34091d0d56c5f07b00043f86b5ece8cb2b4b8854
Thanks